backend/sdk/src/main/java/com/alibaba/higress/sdk/service/DomainServiceImpl.java

```diff
@@ -57,17 +59,8 @@ public DomainServiceImpl(KubernetesClientService kubernetesClientService,
 
     @Override
     public Domain add(Domain domain) {
-        V1ConfigMap domainConfigMap = kubernetesModelConverter.domain2ConfigMap(domain);
-        V1ConfigMap newDomainConfigMap;
-        try {
-            newDomainConfigMap = kubernetesClientService.createConfigMap(domainConfigMap);
-        } catch (ApiException e) {
-            if (e.getCode() == HttpStatus.CONFLICT) {
-                throw new ResourceConflictException();
-            }
-            throw new BusinessException("Error occurs when adding a new domain.", e);
-        }
-        return kubernetesModelConverter.configMap2Domain(newDomainConfigMap);
+        DomainStrategy strategy = getStrategy(domain);
```

Comment:
Whether it is possible to make a unified context judgment between controller and service instead of coupling to business service for policy pattern.

Reply:
Or abstract a layer of Service Proxy layer is also fine.

Reply:
Is this design feasible?
1. create `DomainServiceProxy`：
   ```java
   @Service
   public class DomainServiceProxy {
       private final IngressDomainService ingressDomainService;
       private final GatewayDomainService gatewayDomainService;
   
       public DomainServiceProxy(IngressDomainService ingressDomainService, 
                               GatewayDomainService gatewayDomainService) {
           this.ingressDomainService = ingressDomainService;
           this.gatewayDomainService = gatewayDomainService;
       }
   
       public Domain add(Domain domain) {
           return getDomainService(domain.getIsIngressMode()).add(domain);
       }
   
       private DomainService getDomainService(boolean isIngressMode) {
           return isIngressMode ? ingressDomainService : gatewayDomainService;
       }
   }
   ```

2. Split `DomainServiceImpl` into `IngressDomainServiceImpl` and `GatewayDomainServiceImpl `based on business logic.
   ```java
   @Service
   public class IngressDomainServiceImpl implements DomainService {
       // implementation for Ingress
   }
   @Service
   public class GatewayDomainService implements DomainService {
       // implementation for Gateway
   }
   ```

3. Controller:
   ```java
   public class DomainsController {
   
       @Resource
       private DomainServiceProxy domainServiceProxy; 
   
       @PostMapping
       public ResponseEntity<Response<Domain>> add(@RequestBody Domain domain) {
           return ControllerUtil.buildResponseEntity(domainServiceProxy.add(domain));
       }
   }
   ```